File: java/src/main/java/com/tomoarrow/idv/bff/java/controller/TestBoardController.java
package com.tomoarrow.idv.bff.java.controller;

import com.tomoarrow.idv.bff.java.config.AppProperties;
import java.nio.file.Path;
import java.util.Map;
import org.springframework.core.io.FileSystemResource;
import org.springframework.core.io.Resource;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestBoardController {
    private final AppProperties properties;

    public TestBoardController(AppProperties properties) {
        this.properties = properties;
    }

    @GetMapping("/")
    Map<String, String> hello() {
        return Map.of("message", "Hello World!");
    }

    @GetMapping("/test-board")
    ResponseEntity<Resource> testBoard() {
        Resource resource = new FileSystemResource(Path.of(properties.getTestBoardPath()));
        if (!resource.exists()) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok()
                .contentType(MediaType.TEXT_HTML)
                .body(resource);
    }

    @GetMapping("/test-board/config")
    Map<String, String> testBoardConfig() {
        return Map.of(
                "idvServerUrl", properties.resolvedBaseUrl(),
                "idvAppUrl", properties.getAppUrl() == null ? "" : properties.getAppUrl()
        );
    }
}

File: kotlin/src/main/kotlin/com/tomoarrow/idv/bff/controller/TestBoardController.kt
package com.tomoarrow.idv.bff.controller

import com.tomoarrow.idv.bff.config.AppProperties
import org.springframework.core.io.FileSystemResource
import org.springframework.http.MediaType
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import java.io.File

@RestController
class TestBoardController(
    private val appProperties: AppProperties
) {

    @GetMapping("/test-board", produces = [MediaType.TEXT_HTML_VALUE])
    fun testBoard(): ResponseEntity<FileSystemResource> {
        val file = File(appProperties.testBoardPath)
        if (!file.exists()) {
            return ResponseEntity.notFound().build()
        }
        return ResponseEntity.ok(FileSystemResource(file))
    }

    // test-board.html expects camelCase keys — bypass Jackson SNAKE_CASE by returning raw JSON
    @GetMapping("/test-board/config", produces = [MediaType.APPLICATION_JSON_VALUE])
    fun testBoardConfig(): ResponseEntity<String> {
        val json = """{"googleClientId":"${appProperties.googleClientId}","idvServerUrl":"${appProperties.idvBaseUrl}"}"""
        return ResponseEntity.ok(json)
    }
}
